packages/emptystate/src/react/use-resize-observer.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useRef, useState } from 'react'
+import ResizeObserver from 'resize-observer-polyfill'
+
+const isBrowser = typeof window !== 'undefined'
+
+export default function useResizeObserver() {
+  const ref = useRef()
+
+  const [width, setWidth] = useState(0)
+  const [height, setHeight] = useState(0)
+
+  useEffect(() => {
+    let subscribed = true
+    const { current: el } = ref
+
+    function handleResize(entries) {
+      if (!subscribed || !isBrowser) return
+      const { contentRect } = entries[0]
+
+      setWidth(contentRect.width)
+      setHeight(contentRect.height)
+    }
+
+    const debouncedResize = debounce(150, handleResize)
+    const observer = new ResizeObserver(debouncedResize)
+
+    observer.observe(el)
+
+    return () => {
+      subscribed = false
```

Comment:
Is this where we're setting "unmounted"?

Reply:
yes - not the prettiest thing in the world